src/main/java/seedu/address/model/ModelHistoryManager.java

```diff
@@ -125,40 +129,88 @@ public ModelHistoryRecord undo() throws AlfredModelHistoryException {
     }
 
     /**
-     * Returns a String representing undo-able and redo-able Command History, separated by `=` delimiter.
-     * @return String representing undo-able and redo-able Command History.
+     * Returns a List of CommandRecords representing undo-able and redo-able Command History.
+     * @return List of CommandRecords representing undo-able and redo-able Command History.
      */
-    public ArrayList<CommandRecord> getCommandHistory() throws AlfredModelHistoryException {
+    public ArrayList<CommandRecord> getCommandHistory() {
         ArrayList<CommandRecord> commandHistory = new ArrayList<>();
         //Obtain Redo-able Command History
+        commandHistory.add(CommandRecord.getRedoEndPoint());
         int currentIndex = this.history.indexOf(this.current);
         for (int j = this.history.size() - 1; j > currentIndex; j--) {
```

Comment:
To consider for 1.4 refactoring: Implement history as two arrays/stacks - one for future one for past so it might be easier to manage. but this is optional

Reply:
ok